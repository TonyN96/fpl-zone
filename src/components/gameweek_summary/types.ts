export interface SummaryStatType {
  label: string;
  teamCode?: number;
  playerName?: string;
  value?: string | number;
}
