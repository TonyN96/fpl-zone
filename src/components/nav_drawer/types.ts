export interface MenuItemType {
  id: string;
  label: string;
  type: "nav" | "auth";
  onClick: () => void;
}
