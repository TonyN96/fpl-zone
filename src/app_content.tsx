import React from "react";
import { useQuery } from "react-query";
import { BrowserRouter as Router,Route, Routes } from "react-router-dom";
import { useMediaQuery } from "@mui/material";
import { getAllFixtures, getGameData } from "api/fpl_api_provider";
import {
  AccountPage,
  AnalysisPage,
  FixturesAndResultsPage,
  GameweekLivePage,
  LoginPage,
  MyTeamPage,
  RegisterPage,
  ResetPage
} from "pages";
import PrivateRoute from "private_route";
import { AppData } from "types";

import { Logout } from "components/authentication";
import { Notifier, Startup } from "components/layout";

export const AppDataContext = React.createContext<AppData | null>(null);

export default function AppContent(): JSX.Element {
  // Fetching game data which will be made available throughout the app via context provider
  const {
    data: gameData,
    isError: gameDataIsError,
    error: gameDataError,
    isLoading: gameDataIsLoading
  } = useQuery("game-data", getGameData);

  // Fetching fixture data which will be made available throughout the app via context provider
  const {
    data: fixtureData,
    isError: fixtureDataIsError,
    error: fixtureDataError,
    isLoading: fixtureDataIsLoading
  } = useQuery("all-fixtures", getAllFixtures);

  const isCompact = useMediaQuery("(max-width:1500px)");

  const isLoading = gameDataIsLoading || fixtureDataIsLoading;
  const isError = gameDataIsError || fixtureDataIsError;
  const appData = gameData && fixtureData ? { ...gameData, fixtures: fixtureData, isCompact } : null;

  if (isLoading) {
    // Display loading message if data is still being fetched
    return (
      <Startup>
        <Notifier />
      </Startup>
    );
  } else if (isError) {
    // Display error message if data fetch failed
    const error = gameDataError || fixtureDataError;
    const errorMessage = error instanceof Error ? `: ${error.message}` : ".";

    return (
      <Startup>
        <Notifier type='error' message={`An error has occured: ${errorMessage}`} />
      </Startup>
    );
  }

  // Otherwise, render the app
  return (
    <AppDataContext.Provider value={appData}>
      <Router>
        <Routes>
          <Route path='*' element={<LoginPage />} />
          <Route path='/' element={<LoginPage />} />
          <Route path='/login' element={<LoginPage />} />
          <Route path='/register' element={<RegisterPage />} />
          <Route path='/reset' element={<ResetPage />} />
          <Route path='/logout' element={<Logout />} />
          <Route path='/account' element={<PrivateRoute component={<AccountPage />} />} />
          <Route
            path='/gameweek-live'
            element={<PrivateRoute component={<GameweekLivePage />} />}
          />
          <Route path='/my-team' element={<PrivateRoute component={<MyTeamPage />} />} />
          <Route
            path='/fixtures-and-results'
            element={<PrivateRoute component={<FixturesAndResultsPage />} />}
          />
          <Route path='/analysis' element={<PrivateRoute component={<AnalysisPage />} />} />
        </Routes>
      </Router>
    </AppDataContext.Provider>
  );

}
