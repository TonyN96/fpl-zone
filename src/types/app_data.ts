import { Fixture, GameSettings, Gameweek, Phase, Player, PlayerStat, Position,Team } from "types";

export interface AppData {
  gameweeks: Gameweek[];
  gameSettings: GameSettings;
  phases: Phase[];
  teams: Team[];
  playerCount: number;
  players: Player[];
  playerStats: PlayerStat[];
  positions: Position[];
  fixtures: Fixture[];
  isCompact: boolean;
}
