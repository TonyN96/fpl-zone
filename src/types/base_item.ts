import { Player } from "./player";
import { Team } from "./team";

export type BaseItem = Player | Team;

export const isPlayer = (item: BaseItem): item is Player => {
  return "web_name" in item;
};
