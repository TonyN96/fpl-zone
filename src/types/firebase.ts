export interface FirebaseResponse {
  code: string;
  message: string;
  name: string;
}
