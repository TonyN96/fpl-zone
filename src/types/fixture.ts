export interface Fixture {
  id: number;
  code: number;
  team_h: number;
  team_h_score?: unknown;
  team_a: number;
  team_a_score?: unknown;
  event: number;
  finished: boolean;
  minutes: number;
  provisional_start_time: boolean;
  kickoff_time: Date;
  event_name: string;
  is_home: boolean;
  difficulty: number;
}
