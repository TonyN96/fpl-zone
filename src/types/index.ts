export * from "./app_data";
export * from "./fixture";
export * from "./game_data";
export * from "./game_settings";
export * from "./gameweek";
export * from "./league";
export * from "./lineup";
export * from "./phase";
export * from "./player";
export * from "./player_past";
export * from "./player_performance";
export * from "./player_stat";
export * from "./player_summary";
export * from "./position";
export * from "./region";
export * from "./team";
export * from "./team_data";
export * from "./team_history";
export * from "./team_picks";
export * from "./transfer";
