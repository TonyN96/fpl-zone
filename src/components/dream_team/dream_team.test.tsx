import React from "react";
import { render } from "@testing-library/react";
import DreamTeam from "./dream_team";
import { AppDataContext } from "app_content";
import { mockElementStats, mockPlayers } from "test/fixture-data";
import { GameData } from "types";

describe("Dream Team Tests", () => {
  const appData = {
    gameData: {
      element_types: mockElementStats,
      elements: mockPlayers,
      
    } as GameData
  }

  const createComponent = (): JSX.Element => {
    return (<AppDataContext.Provider value={appData}><DreamTeam /></AppDataContext.Provider>);
  };

  it("Snapshot test", () => {
    const { asFragment } = render(createComponent());
    expect(asFragment()).toMatchSnapshot();
  });
});
