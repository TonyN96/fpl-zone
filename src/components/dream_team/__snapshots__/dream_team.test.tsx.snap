// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dream Team Tests Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-12kpaeh"
  >
    <div
      class="MuiBox-root css-1bxf5j8"
      data-testid="first-xi-players"
    >
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
      <div
        class="MuiBox-root css-geg32f"
      />
    </div>
    <div
      class="MuiBox-root css-1alt9xp"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Bench
      </h5>
      <div
        class="MuiBox-root css-1doxykz"
        data-testid="bench-players"
      />
    </div>
  </div>
</DocumentFragment>
`;
